compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs

```diff
@@ -263,14 +367,53 @@ impl<'cx, 'tcx> LexicalResolver<'cx, 'tcx> {
         }
 
         match *b_data {
+            VarValue::Empty(empty_ui) => {
+                let lub = match *a_region {
+                    ReLateBound(..) | ReErased => {
+                        bug!("cannot relate region: {:?}", a_region);
+                    }
+
+                    ReVar(v_id) => {
+                        span_bug!(
+                            self.var_infos[v_id].origin.span(),
+                            "expand_node invoked with non-concrete regions: {:?}",
+                            a_region,
+                        );
+                    }
+
+                    ReStatic => {
+                        // nothing lives longer than `'static`
+                        self.tcx().lifetimes.re_static
+                    }
+
+                    ReEarlyBound(_) | ReFree(_) => {
+                        // All empty regions are less than early-bound, free,
+                        // and scope regions.
+                        a_region
+                    }
+
+                    RePlaceholder(placeholder) => {
+                        // If this empty region is from a universe that can
+                        // name the placeholder, then the placeholder is
+                        // larger; otherwise, the only ancestor is `'static`.
+                        if empty_ui.can_name(placeholder.universe) {
+                            self.tcx().mk_region(RePlaceholder(placeholder))
+                        } else {
+                            self.tcx().lifetimes.re_static
+                        }
+                    }
+                };
```

Comment:
except for placeholder handling, this is exactly the same as the `lub` computation in `fn expansion`. Maybe pull it out into a function (that returns `Result<Region, Placeholder>`?)